### PR TITLE
Feat : Login Authentication

### PR DESCRIPTION
**Pull request by anuj-kumary**
Added Login Authentication
- For now, log in via only hardcoded test credentials.

### CODE REVIEW

File: eagle-stores/src/components/NavBar/NavBar.jsx [hunk 1]
```diff
@@ -1,56 +1,58 @@
...
+            className='search__box'
+            type='search'
+            placeholder='Search with name or categories'
           />
         </ul>
 
-        <ul className="navbar__right">
+        <ul className='navbar__right'>
           <div>
-            <Link to="/product" href="./Pages/Authentication/login.html">
```
Comment (Rohanmond) [inline eagle-stores/src/components/NavBar/NavBar.jsx:39]: delete href

File: eagle-stores/src/components/NavBar/NavBar.jsx [hunk 2]
```diff
@@ -1,56 +1,58 @@
...
-              Explore
+            <Link to='/product'>Explore</Link>
+          </div>
+          <div className='badge'>
+            <Link to='/login' href='/Pages/Cart Page/cart-page.html'>
```
Comment (Rohanmond) [inline eagle-stores/src/components/NavBar/NavBar.jsx]: delete href

File: eagle-stores/src/context/auth/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+  const localStorageToken = JSON.parse(localStorage.getItem('login'));
+  const [token, setToken] = useState(localStorageToken?.token);
+  const [user, setUser] = useState(null);
+
+  const loginHandler = async (email, password) => {
+    try {
+      const resp = await axios.post('/api/auth/login', {
```
Comment (Rohanmond) [inline eagle-stores/src/context/auth/auth-context.js]: better to make a services.js file and defined these req as a function there and then use those calling functions here
i.e const LoginService=async()=>{
 return axios.post("api")
}

File: eagle-stores/src/pages/Authentication/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            className='btn btn__primary'
+            onClick={() => loginHandler(loginUser.email, loginUser.password)}
+          >
+            Login
+          </button>
+        </div>
+        <div className='text__center'>
+          <a className='login__link fw__400' href='./signup.html'>
+            Create new account
+          </a>
```
Comment (romabulani) [inline eagle-stores/src/pages/Authentication/Login.jsx]: As html pages, dont exist, you can provide href as `#` for now or use  Link

**Review by Rohanmond — APPROVED** [review]

File: eagle-stores/src/pages/Authentication/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            value={loginUser.password}
+          />
+        </div>
+        <div className='input'>
+          <label className='input__checkbox'></label>
+          <input type='checkbox' />
+          <span className='text'>Remember Me</span>
+          <a href='./forget.html' className='auth__forget'>
+            Forget your Password?
+          </a>
```
Comment (romabulani) [inline eagle-stores/src/pages/Authentication/Login.jsx]: Here also, same for html

**Review by romabulani — APPROVED** [review]
- Very clean code
- Liked that you are using `' '` instead of `" "`
- suggested a couple of changes if you have bandwidth, looks good to merge otherwise

File: eagle-stores/src/context/auth/auth-context.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        email,
+        password,
+      });
+      console.log(resp.data);
```
Comment (manish591) [inline eagle-stores/src/context/auth/auth-context.js:17]: remove console

**Review by manish591 — APPROVED** [review]
Looking good,

File: eagle-stores/src/components/NavBar/NavBar.jsx (hunk 1 above)
Comment (anuj-kumary) [reply eagle-stores/src/components/NavBar/NavBar.jsx:39]: Done

File: eagle-stores/src/components/NavBar/NavBar.jsx (hunk 2 above)
Comment (anuj-kumary) [reply eagle-stores/src/components/NavBar/NavBar.jsx]: Done

File: eagle-stores/src/pages/Authentication/Login.jsx (hunk 4 above)
Comment (anuj-kumary) [reply eagle-stores/src/pages/Authentication/Login.jsx]: Thanks for the suggestion. I will work on it

File: eagle-stores/src/context/auth/auth-context.js (hunk 6 above)
Comment (anuj-kumary) [reply eagle-stores/src/context/auth/auth-context.js:17]: done

File: eagle-stores/src/components/NavBar/NavBar.jsx (hunk 1 above)
Comment (varnita619) [inline eagle-stores/src/components/NavBar/NavBar.jsx:39]: remove href as it is of no use after applying Link

File: eagle-stores/src/components/NavBar/NavBar.jsx [hunk 7]
```diff
@@ -1,56 +1,58 @@
...
+              <i className='badge__icon fas fa-user'></i>
             </Link>
           </div>
-
-          <div className="badge">
-            <a href="/Pages/Cart Page/cart-page.html">
```
Comment (varnita619) [inline eagle-stores/src/components/NavBar/NavBar.jsx:45]: change this anchor tag to Link as there is no html file now

File: eagle-stores/src/context/auth/auth-context.js (hunk 3 above)
Comment (anuj-kumary) [reply eagle-stores/src/context/auth/auth-context.js]: Thanks for the suggestion i will definitely make these changes

File: eagle-stores/src/pages/Authentication/Auth.css [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    color: var(--primary-color);
+}
+
+.input__checkbox {
+    vertical-align: middle;
+}
+
+.text {
+    vertical-align: text-bottom;
+}
```
Comment (varnita619) [inline eagle-stores/src/pages/Authentication/Auth.css:58]: you can add a new line at the end of each file

**Review by varnita619 — APPROVED** [review]

File: eagle-stores/src/components/NavBar/NavBar.jsx (hunk 7 above)
Comment (anuj-kumary) [reply eagle-stores/src/components/NavBar/NavBar.jsx:45]: Done

**Pull request merged**